operators/volume_loader/nrrd_loader.cpp

```diff
@@ -148,56 +148,96 @@ bool load_nrrd(const std::string& file_name, Volume& volume) {
           }
         }
         volume.SetOrientation(orientation);
+      } else if (parameter == "spaceorigin") {
+        auto space_origin = parse_vector(value);
+        std::copy_n(space_origin.begin(), 3, volume.space_origin_.begin());
+      } else if (parameter == "spacedirections") {
+        auto values = split_string_by_space(value);
+        for (const auto& value : values) {
+          auto space_directions = parse_vector(value);
+          std::array<double, 3> space_direction = {0.0, 0.0, 0.0};
+          std::copy_n(space_directions.begin(), 3, space_direction.begin());
+          volume.space_directions_.push_back(space_direction);
+        }
       }
     }
   }
 
+  int byte_skip = 0;
+  {
+    std::ifstream file;
+    file.open(file_name, std::ios::in);
+    if (!file.is_open()) {
+      holoscan::log_error("NRRD could not open {}", file_name);
+      return false;
+    }
+    std::string line;
+    while (std::getline(file, line)) {
+      if (file.tellg() != -1) { byte_skip = file.tellg(); }
+    }
+  }
```

Comment:
Could we integrate this code into the loop above? Opening the file again and reading the header to get the header size seems to be inefficient.

Reply:
Please review the changes; I also refactored the function into smaller ones. Thank you!